sentinelhub/data_sources.py

```diff
@@ -0,0 +1,522 @@
+"""
+Module defining data sources
+"""
+import re
+import warnings
+from enum import Enum, EnumMeta
+from typing import Tuple
+from dataclasses import dataclass, asdict
+
+from aenum import extend_enum
+
+from .config import SHConfig
+from .constants import ServiceUrl
+from .exceptions import SHDeprecationWarning
+
+
+class _Source:
+    """ Types of satellite sources
+    """
+    SENTINEL2 = 'Sentinel-2'
+    SENTINEL1 = 'Sentinel-1'
+    LANDSAT8 = 'Landsat 8'
+    MODIS = 'MODIS'
+    DEM = 'Mapzen DEM'
+    BYOC = 'BYOC'
+    BATCH = 'BATCH'
+    LANDSAT5 = 'Landsat 5'
+    LANDSAT7 = 'Landsat 7'
+    SENTINEL3 = 'Sentinel-3'
+    SENTINEL5P = 'Sentinel-5P'
+    ENVISAT_MERIS = 'Envisat Meris'
+
+
+class _SensorType:
+    """ Types of satellite sensors
+    """
+    # pylint: disable=invalid-name
+    MSI = 'MSI'
+    IW = 'IW'
+    EW = 'EW'
+    OLCI = 'OLCI'
+    SLSTR = 'SLSTR'
+    TROPOMI = 'TROPOMI'
+
+
+class _ProcessingLevel:
+    """ Types of processing level
+    """
+    # pylint: disable=invalid-name
+    L1B = 'L1B'
+    L1C = 'L1C'
+    L2A = 'L2A'
+    L3B = 'L3B'
+    L2 = 'L2'
+    GRD = 'GRD'
+    MCD43A4 = 'MCD43A4'
+
+
+class _Polarization:
+    """ Types of SAR polarizations
+    """
+    # pylint: disable=invalid-name
+    DV = 'DV'
+    DH = 'DH'
+    SV = 'SV'
+    SH = 'SH'
+
+
+class _Resolution:
+    """ Types of product resolution (specific to Sentinel-1 sources)
+    """
+    MEDIUM = 'MEDIUM'
+    HIGH = 'HIGH'
+
+
+class OrbitDirection:
+    """ Types of orbit directions
+    """
+    ASCENDING = 'ASCENDING'
+    DESCENDING = 'DESCENDING'
+    BOTH = 'BOTH'
+
+
+class _Bands:
+    """ Different collections of bands
+    """
+    SENTINEL2_L1C = ('B01', 'B02', 'B03', 'B04', 'B05', 'B06', 'B07', 'B08', 'B8A', 'B09', 'B10', 'B11', 'B12')
+    SENTINEL2_L2A = ('B01', 'B02', 'B03', 'B04', 'B05', 'B06', 'B07', 'B08', 'B8A', 'B09', 'B11', 'B12')
+    SENTINEL3_OLCI = tuple(f'B{str(index).zfill(2)}' for index in range(1, 22))
+    SENTINEL3_SLSTR = ('S1', 'S2', 'S3', 'S4', 'S5', 'S6', 'S7', 'S8', 'S9', 'F1', 'F2')
+    LANDSAT8 = ('B01', 'B02', 'B03', 'B04', 'B05', 'B06', 'B07', 'B08', 'B09', 'B10', 'B11')
+    DEM = ('DEM',)
+    MODIS = ('B01', 'B02', 'B03', 'B04', 'B05', 'B06', 'B07')
+    SENTINEL1_IW = ('VV', 'VH')
```

Comment:
There are more polarisations available for acquisition modes [docs](https://docs.sentinel-hub.com/api/latest/data/sentinel-1-grd/#acquisitionmode). How do these link to the available bands?

Reply:
I would leave the other polarizations and bands for users to define themselves. Here is just wanted to define the most commonly used ones.